optimum/intel/neural_compressor/trainer.py

```diff
@@ -59,40 +60,93 @@
-        trial: Union["optuna.Trial", Dict[str, Any]] = None,
-        ignore_keys_for_eval: Optional[List[str]] = None,
-        **kwargs,
+        model: Union[PreTrainedModel, torch.nn.Module] = None,
+        args: TrainingArguments = None,
+        data_collator: Optional[DataCollator] = None,
+        train_dataset: Optional[Dataset] = None,
+        eval_dataset: Optional[Dataset] = None,
+        tokenizer: Optional[PreTrainedTokenizerBase] = None,
+        model_init: Callable[[], PreTrainedModel] = None,
+        compute_metrics: Optional[Callable[[EvalPrediction], Dict]] = None,
+        callbacks: Optional[List[TrainerCallback]] = None,
+        optimizers: Tuple[torch.optim.Optimizer, torch.optim.lr_scheduler.LambdaLR] = (None, None),
+        preprocess_logits_for_metrics: Callable[[torch.Tensor, torch.Tensor], torch.Tensor] = None,
+        quantization_config: Optional[_BaseQuantizationConfig] = None,
+        pruning_config: Optional[_BaseQuantizationConfig] = None,
+        distillation_config: Optional[_BaseQuantizationConfig] = None,
+        task: Optional[str] = None,
+        save_onnx_model: bool = False,
     ):
 
-        """
-        Main training entry point.
-        Args:
-            agent (:obj:`Component`, *optional*):
-                Component object containing the compression objects to apply during the training process.
-            resume_from_checkpoint (`str` or `bool`, *optional*):
-                If a `str`, local path to a saved checkpoint as saved by a previous instance of [`IncTrainer`]. If a
-                `bool` and equals `True`, load the last checkpoint in *args.output_dir* as saved by a previous instance
-                of [`IncTrainer`]. If present, training will resume from the model/optimizer/scheduler states loaded here.
-            trial (`optuna.Trial` or `Dict[str, Any]`, *optional*):
-                The trial run or the hyperparameter dictionary for hyperparameter search.
-            ignore_keys_for_eval (`List[str]`, *optional*):
-                A list of keys in the output of your model (if it is a dictionary) that should be ignored when
-                gathering predictions for evaluation during the training.
-            kwargs:
-                Additional keyword arguments used to hide deprecated arguments
-        """
-        self.agent = agent
-        self.args.use_ipex = False
+        super().__init__(
+            model,
+            args,
+            data_collator,
+            train_dataset,
+            eval_dataset,
+            tokenizer,
+            model_init,
+            compute_metrics,
+            callbacks,
+            optimizers,
+            preprocess_logits_for_metrics,
+        )
+
+        inc_config = []
+        self.task = task
+        self.quantization_config = quantization_config
+        self.pruning_config = pruning_config
+        self.distillation_config = distillation_config
+
+        self.compression_controller = None

```

Comment:
INC used `compression_manager` class to store compression related info. looks compression_controller is the concept of NNCF. suggest to change the variable name to compression_manager to avoid confusing.

Reply:
Sure I'll change it !